external/hbase/src/main/scala/io/gearpump/external/hbase/HBaseSecurityUtil.scala

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.gearpump.external.hbase
+
+import java.io.File
+
+import io.gearpump.util.FileUtils
+import io.gearpump.cluster.UserConfig
+import io.gearpump.util.{LogUtil, Constants}
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.hbase.client.{ConnectionFactory, Connection}
+import org.apache.hadoop.security.UserGroupInformation
+import org.slf4j.Logger
+
+object HBaseSecurityUtil {
+  val LOG: Logger = LogUtil.getLogger(getClass)
+
+  def getConnection(userConfig: UserConfig, configuration: Configuration): Connection = {
+    if(UserGroupInformation.isSecurityEnabled) {
+      val principal = userConfig.getString(Constants.GEARPUMP_KERBEROS_PRINCIPAL)
+      val keytabContent = userConfig.getBytes(Constants.GEARPUMP_KEYTAB_FILE)
+      if(principal.isEmpty || keytabContent.isEmpty) {
+        val errorMsg = s"HBase is security enabled, user should provide kerberos principal in " +
+          s"${Constants.GEARPUMP_KERBEROS_PRINCIPAL} and keytab file in ${Constants.GEARPUMP_KEYTAB_FILE}"
+        throw new Exception(errorMsg)
+      }
```

Comment:
It is better, but not solving the problem.

With no access control, it is still possble for other user to access it.